### PR TITLE
Add-Immich

### PR DESCRIPTION
**Pull request by RaneyDazed**
# Description

Immich is a self hosted photo and video backup solution, and allows for a decent amount of customization. 

## Docs

https://immich.app/docs/overview/introduction

### Link to GH/Docker image used in this role

https://github.com/imagegenius/docker-immich

### Link to GH/Original (non monolithic) version, and the devs of the container/app "Immich"

https://github.com/immich-app/immich

## Docs? 

I will make a doc page with some basic info on Immich.

# How Has This Been Tested?

I've tested this on 2 different servers so far. I'll attempt to get more people to give it a whirl as much as I can.

- [x] Installed on 2 separate servers

### CODE REVIEW

File: roles/immich/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+# Commands
+immich_docker_commands_default: []
+immich_docker_commands_custom: []
+immich_docker_commands: "{{ immich_docker_commands_default
+                            + immich_docker_commands_custom }}"
+
+# Volumes
+immich_docker_volumes_default:
+  - "{{ immich_paths_location }}:/config"
+  - "{{ immich_paths_location }}/machine-learning:/config/machine-learning"
```
Comment (owine) [inline roles/immich/defaults/main.yml:103]: Is this volume mount needed? Wouldn't the `/config` one cover it?

Comment (RaneyDazed) [reply roles/immich/defaults/main.yml:103]: Heya @owine, yes I think it is. I may be misunderstanding? but either way, even if its not needed, the container makes that dir by default. So if figured we might as well make the role create it so its not owned by root.

Comment (saltydk) [reply roles/immich/defaults/main.yml:103]: You can create the folder, which makes sense, the mounting does not.

Comment (RaneyDazed) [reply roles/immich/defaults/main.yml:103]: I'll test it without real quick and report back.

Comment (RaneyDazed) [reply roles/immich/defaults/main.yml]: Yeah that was dumb. :p tyvm fellas

**Pull request merged**